allofplos/transformations.py

```diff
@@ -100,9 +100,8 @@ def filename_to_doi(filename):
         doi = PREFIX + article
     elif validate_filename(filename):
         doi = PREFIX + os.path.splitext(filename)[0]
-    # NOTE: A filename should never validate as a DOI, so the next elif is wrong.
-    elif validate_doi(filename):
-        doi = filename
+    else:
+        doi = ''
     return doi
```

Comment:
You should validate the doi before returning it.

Reply:
Do we have a custom doi validation error? If not… we should.

Reply:
BTW, you have the same problem with doi_to_path

https://github.com/PLOS/allofplos/pull/88/files?diff=unified#diff-ad065d41199b64627a6f15ef79720843R181

```
     if directory is None:
         directory = get_corpus_dir()
     if doi.startswith(ANNOTATION_DOI) and validate_doi(doi):
         article_file = os.path.join(directory, "plos.correction." + doi.split('/')[-1] + SUFFIX_LOWER)
     elif validate_doi(doi):
         article_file = os.path.join(directory, doi.lstrip(PREFIX) + SUFFIX_LOWER)
     # NOTE: The following check is weird, a DOI should never validate as a file name.
     elif validate_filename(doi):
         article_file = doi
     return article_file
```

it should have an else case.